tests/test_flow/fakes.py

```diff
@@ -166,8 +169,40 @@ def rm(self, url, recursive=False):
             del self._files_by_url[url]
 
 
+def move_files_in_dir(src_dir, dst_dir):
+    """
+    Move all files from one directory to another.
+    """
+    filenames = os.listdir(src_dir)
+    for filename in filenames:
+        shutil.move(os.path.join(src_dir, filename), dst_dir)
+
+
+@contextmanager
+def temporary_clean_dir(target_dir):
+    """
+    Temporarily remove all files in a directory for the duration of a context.
+    Once the context ends, the directory is restored to its original state. That
+    means any new file created during the duration of the context are deleted.
+    """
+    with TemporaryDirectory() as tmp_backup_dir:
```

Comment:
Oh, and if we only clean cache dir, then we can also use `tmp_path` here as well instead of using `TemporaryDirectory`.

Reply:
I am using `TemporaryDirectory` here because it is much simpler. What do you think?